homeassistant/components/binary_sensor/xiaomi_aqara.py

```diff
@@ -311,6 +316,48 @@ def parse_data(self, data, raw_data):
             return False
 
 
+class XiaomiVibration(XiaomiBinarySensor):
+    """Representation of a Xiaomi Vibration Sensor."""
+
+    def __init__(self, device, name, data_key, hass, xiaomi_hub):
+        """Initialize the XiaomiVibration."""
+        self._hass = hass
+        self._last_action = None
+        XiaomiBinarySensor.__init__(self, device, name, xiaomi_hub,
+                                    data_key, None)
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        attrs = {ATTR_LAST_ACTION: self._last_action}
+        attrs.update(super().device_state_attributes)
+        return attrs
+
+    def parse_data(self, data, raw_data):
+        """Parse data sent by gateway."""
+        value = data.get(self._data_key)
+        if value is None:
+            return False
+
+        if value == 'vibrate':
+            movement_type = 'vibrate'
+        elif value == 'tilt':
+            movement_type = 'tilt'
+        elif value == "free_fall":
+            movement_type = 'free_fall'
+        else:
+            _LOGGER.warning("Unsupported movement_type detected: %s", value)
+            return False
```

Comment:
Replace it with 
```

if value not in ('vibrate', 'tilt',  'free_fall'):
            _LOGGER.warning("Unsupported movement_type detected: %s", value)
            return False


```

Reply:
I'm not sure why we would want an "if" statement enumerating all three of those conditions and then supplying an "else" to handle anything unexpected... and then also putting all three conditions in the "else"... it seems redundant. But, the change is simple enough, so I can make it. 

The problem now is that, in the event that we somehow make it to the "else" with one of those enumerated values being set (which should be impossible but, if we're checking for it, that means we think it isn't) then the "if" statement in the "else" will not catch and we'll progress through the code without movement_type being set which will surely break. Again, it should be impossible to get there and NOT have that condition be true, it just seems weird. But, again, Python isn't my strongest language... so... maybe there's some hidden reason for it.

The code now looks like this:

````
    def parse_data(self, data, raw_data):
        """Parse data sent by gateway."""
        value = data.get(self._data_key)
        if value is None:
            return False

        if value == 'vibrate':
            movement_type = 'vibrate'
        elif value == 'tilt':
            movement_type = 'tilt'
        elif value == "free_fall":
            movement_type = 'free_fall'
        else:
            if value not in ('vibrate', 'tilt',  'free_fall'):
                _LOGGER.warning("Unsupported movement_type detected: %s", value)
                return False

        self._hass.bus.fire('movement', {
            'entity_id': self.entity_id,
            'movement_type': movement_type
        })
        self._last_action = movement_type

        return True
````

Reply:
Replace line 339-350 with 
```
if value not in ('vibrate', 'tilt',  'free_fall'):
            _LOGGER.warning("Unsupported movement_type detected: %s", value)
            return False
```